analyses/cell-type-ewings/exploratory_analysis/06-tumor-cell-state-assignment.Rmd

```diff
@@ -0,0 +1,629 @@
+
+# add clusters to tumor cells df for plotting
+tumor_cells_df <- tumor_cells_df |> 
+  dplyr::left_join(subcluster_df, by = c("library_id", "barcodes"))
+```
+
+The first thing we will do is plot the cluster assignments on the UMAP. 
+
+```{r}
+library_ids |>
+  purrr::map(\(id){
+    tumor_cells_df |>
+      dplyr::filter(library_id == id) |>
+      ggplot(aes(x = UMAP1, y = UMAP2, color = sub_cluster)) +
+      geom_point(alpha = 0.5, size = 0.1) +
+      theme(
+        aspect.ratio = 1
+      ) +
+      labs(title = id) +
+      scale_color_brewer(palette = "Dark2") +
+      guides(color = guide_legend(override.aes = list(alpha = 1, size = 1.5)))
+  })
+```
+
+Now we will look at the marker gene expression across all clusters for each sample, first using a density plot and then heatmaps. 
+
+```{r, fig.height=10}
+library_ids |> 
+  purrr::map(\(id){
+    
+    density_df <- tumor_cells_df |> 
+      dplyr::filter(library_id == id)
+    
+    marker_gene_cols |>
+    purrr::map(\(column){
+      plot_density(
+        density_df,
+        column,
+        "sub_cluster"
+      ) 
+    }) |>
+    patchwork::wrap_plots(ncol = 1) +
+      patchwork::plot_annotation(id)
+    
+  })
+```
+
+In all of the heatmaps below, the `annotation` column is the `sub_cluster`. 
+
+```{r}
+library_ids |> 
+  purrr::map(\(id){
+    
+    filtered_df <- tumor_cells_df |> 
+      dplyr::filter(library_id == id)
+    
+    full_celltype_heatmap(filtered_df, marker_gene_cols, "sub_cluster")
+    
+  })
+
+```
+
+Let's make the same heatmap, but remove the column clustering.  
+
+```{r}
+library_ids |> 
+  purrr::map(\(id){
+    
+    filtered_df <- tumor_cells_df |> 
+      dplyr::filter(library_id == id) |> 
+      dplyr::arrange(sub_cluster)
+    
+    full_celltype_heatmap(filtered_df, marker_gene_cols, "sub_cluster", cluster_columns = FALSE)
+    
+  })
+```
+
+It looks like for both samples we have some clear division between EWS high and EWS low cells. 
+Additionally, there are some cells with really strong expression of the EWS high markers and others that look like the genes are expressed but to a lesser degree. 
+This fits with the findings from the literature that `EWS-FLI1` expression lies on a continuum. 
+
+Between these two plots I think we can try and group cells into EWS low, EWS middle, and EWS high cells. 
+
+The last thing we'll do is look at any cells that could be marked as "proliferative". 
+`SCPCL000822` doesn't appear to have very much expression of the proliferative markers, but `SCPCL000824` does seem to have one cluster, cluster 2, that has expression of the proliferative cells. 
+Perhaps we should label those cells separately as `EWS-high-proliferative`. 
+
+Based on this I would make the following assignments: 
+
+`SCPCL000822`: 
+`EWS-FLI1` low  - 2
+`EWS-FLI1` middle - 1 & 4
+`EWS-FLI1` high - 3 
+
+`SCPCL000824` : 
+`EWS-FLI1` low - 1
+`EWS-FLI1` middle - 6 
+`EWS-FLI1` high - 3, 4, 5, 7
+`EWS-FLI1` high proliferative - 2
```

Comment:
Forgot to add this comment before - this is each appearing on one line so let's actually make it bullets

```suggestion
Based on this I would make the following assignments: 

`SCPCL000822`: 

* `EWS-FLI1` low  - 2
* `EWS-FLI1` middle - 1 & 4
* `EWS-FLI1` high - 3 

`SCPCL000824` : 

* `EWS-FLI1` low - 1
* `EWS-FLI1` middle - 6 
* `EWS-FLI1` high - 3, 4, 5, 7
* `EWS-FLI1` high proliferative - 2
```